smarts/core/road_map.py

```diff
@@ -383,7 +399,15 @@ def outgoing_roads(self) -> List[RoadMap.Road]:
         def oncoming_roads_at_point(self, point: Point) -> List[RoadMap.Road]:
             """Returns a list of nearby roads to point that are (roughly)
             parallel to this one but have lanes that go in the opposite direction."""
-            raise NotImplementedError()
+            result = []
+            for lane in self.lanes:
+                offset = lane.to_lane_coord(point).s
+                result += [
+                    ol.road
+                    for ol in lane.oncoming_lanes_at_offset(offset)
+                    if ol.road != self
+                ]
+            return result
```

Comment:
Yeah this was moved here, since the current formats have the same implementation. But I think its better to move them to their respective road network class for now.